src/permission.rs

```diff
@@ -26,45 +22,42 @@ struct Perm {
 
 type PermList = Vec<Perm>;
 
+static IS_ROLE_PERM: LazyLock<Regex> =
+    LazyLock::new(|| Regex::new("([^:]+)::roles::(.*)").unwrap());
+
 async fn fetch_perm(perm_uri: &PermUri) -> Option<PermList> {
-    let client = Client::new();
-    let res = match client.get(perm_uri.uri.clone()).await {
-        Ok(res) => res,
-        Err(e) => {
-            error!("fail to fetch {:?}: {}", perm_uri, e);
-            return None;
-        }
-    };
-    let body = match hyper::body::aggregate(res).await {
-        Ok(body) => body,
-        Err(e) => {
-            error!("fail to fetch {:?}: {}", perm_uri, e);
-            return None;
-        }
-    };
-    match serde_json::from_reader(body.reader()) {
-        Ok(json) => json,
-        Err(e) => {
-            error!("fail to fetch {:?}: {}", perm_uri, e);
-            None
-        }
-    }
+    let client = Client::builder(TokioExecutor::new()).build_http::<Full<Bytes>>();
+
+    let res = client
+        .get(perm_uri.uri.clone())
+        .await
+        .inspect_err(|e| error!("fail to fetch {perm_uri:?}: {e}"))
+        .ok()?;
+
+    let body: BytesMut = res
+        .into_data_stream()
+        .try_collect()
+        .await
+        .inspect_err(|e| error!("fail to fetch {perm_uri:?}: {e}"))
+        .ok()?;
```

Comment:
This part _might_ be slightly less performant as it collects data in a single buffer, but I think serde is slower on a generic `Read` object which might make this actually faster.